ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStore.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.cloud;
+
+import com.azure.storage.blob.BlobContainerClient;
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.cloud.azure.AzureCloudConfig;
+import com.github.ambry.cloud.azure.AzureMetrics;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.ReplicaState;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.replication.FindToken;
+import com.github.ambry.store.FindInfo;
+import com.github.ambry.store.MessageInfo;
+import com.github.ambry.store.MessageWriteSet;
+import com.github.ambry.store.Store;
+import com.github.ambry.store.StoreException;
+import com.github.ambry.store.StoreGetOptions;
+import com.github.ambry.store.StoreInfo;
+import com.github.ambry.store.StoreKey;
+import com.github.ambry.store.StoreStats;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Implements methods to write {@link com.github.ambry.clustermap.AmbryPartition} in Azure blob storage.
+ */
+public class AzureStore implements Store {
```

Comment:
This class will directly talk to Azure and use Azure APIs. In V2, there is no more CloudDestination and I highly doubt we will ever use S3 or GCP. If that happens, then we can refactor the code but I don't want to create those destination classes and objects at the expense of current OKR in the anticipation of an extremely rare possibility.

Reply:
Thanks Arun for the explanation. I second Arun's opinion. 
1. The current architecture to separate the generic cloud vs Azure implement makes sense in the design point of view.
2. And since V1 and V2 will co-exist for quite some time, let's follow the previous naming conversion. Otherwise, the person who familiar with the V1 will get very confused.